### PR TITLE
Fix issues with Carthage build by reintroducing manually-managed Xcode project

### PR DESCRIPTION
**Pull request by calda**
It looks like Carthage has issues (#1492, #1495) with the setup we're using, where the `Lottie` Carthage framework re-exports an underlying `LottieCore` SPM dependency. The best way to fix this seems to be to go back to using a manually-managed Xcode project (like before #1427).

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -68,8 +68,23 @@ namespace :test do
...
+      sh 'rm -rf ~/Library/Caches/org.carthage.CarthageKit/DerivedData'
+
       # Build a test app that imports and uses the LottieCarthage framework
-      sh 'xcodebuild build -scheme CarthageTest -destination "platform=iOS Simulator,name=iPhone 8"'
+      xcodebuild('build -scheme CarthageTest -destination "platform=iOS Simulator,name=iPhone 8"')
+    end
+  end
+
+  desc 'Tests Swift Package Manager support'
+  task :spm do
```
Comment (calda) [inline Rakefile:82]: The `build:package` commands no longer build the targets generated by `Package.swift`, so this adds a new separate `test:spm` flow that validates the targets in `Package.swift`.

File: Lottie.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -0,0 +1,2502 @@
+// !$*UTF8*$!
+{
+	archiveVersion = 1;
+	classes = {
+	};
+	objectVersion = 55;
+	objects = {
+
+/* Begin PBXBuildFile section */
```
Comment (calda) [inline Lottie.xcodeproj/project.pbxproj]: 🥲

Comment (calda) [inline Lottie.xcodeproj/project.pbxproj]: 🥲

File: Rakefile [hunk 3]
```diff
@@ -68,8 +68,23 @@ namespace :test do
       # Build the LottieCarthage framework scheme
       sh 'carthage build --use-xcframeworks'
 
+      # Delete Carthage's derived data to verify that the built .xcframework doesn't depend on any
+      # side effects from building on this specific machine.
+      # https://github.com/airbnb/lottie-ios/issues/1492
+      sh 'rm -rf ~/Library/Caches/org.carthage.CarthageKit/DerivedData'
```
Comment (calda) [inline Rakefile:74]: This changes causes the test to fail with the previous project setup (#1495)

**Review by erichoracek — APPROVED** [review]
@calda LGTM!

**Pull request merged**